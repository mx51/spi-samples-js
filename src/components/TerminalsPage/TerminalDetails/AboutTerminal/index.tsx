import React from 'react';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import {
  terminalConfigurationsPartOne,
  terminalConfigurationsPartTwo,
} from '../../../../definitions/constants/terminalConfigs';
import { useAppDispatch } from '../../../../redux/hooks';
import { handleUnPairClick } from '../../../../utils/common/pair/pairStatusHelpers';
import useStyles from './index.styles';
import StatusBox from './StatusBox';

export default function AboutTerminal({ receiptToggle, setReceiptToggle, terminal }: Any): React.ReactElement {
  const dispatch = useAppDispatch();
  const classes = useStyles();

  const handleReceiptPanelDisplay = () => {
    setReceiptToggle(!receiptToggle);
  };

  return (
    <>
      <Grid container>
        <Grid item className={(classes.fullWidth, classes.sectionSpacing)}>
          <Typography variant="h6" component="h1">
            Terminal information
          </Typography>
          <Typography className={classes.text}>Information about this terminal</Typography>
          <Grid container className={classes.detailRow}>
            <Grid item md={4} xs={12}>
              <Typography>Pairing status</Typography>
            </Grid>
            <Grid item md={8} xs={12}>
              {StatusBox(classes, terminal?.status)}
              <Button className={classes.actionButton} onClick={handleReceiptPanelDisplay}>
                Settlement
              </Button>
              <Button className={classes.actionButton} onClick={handleReceiptPanelDisplay}>
                Settlement enquiry
              </Button>
              <Button className={classes.actionButton} onClick={() => handleUnPairClick(dispatch, terminal.id)}>
                Unpair terminal
              </Button>
            </Grid>
          </Grid>
          {terminalConfigurationsPartOne(terminal).map(({ title, content }) => (
            <Grid className={classes.detailRow} container key={title}>
              <Grid item md={4} xs={12}>
                <Typography>{title}</Typography>
              </Grid>
              <Grid item md={8} xs={12}>
                <Typography className={classes.text}>{content}</Typography>
              </Grid>
            </Grid>
          ))}
        </Grid>
        <Grid item className={(classes.fullWidth, classes.sectionSpacing)}>
          <Typography variant="h6" component="h1">
            Pairing configuration
          </Typography>
          <Typography className={classes.text}>
            To update the Pairing configuration you need to unpair this terminal using the Unpair button above
          </Typography>
          {terminalConfigurationsPartTwo(terminal).map(({ title, content }) => (
            <Grid className={classes.detailRow} container key={title}>
              <Grid item md={4} xs={12}>
                <Typography>{title}</Typography>
              </Grid>
              <Grid item md={8} xs={12}>
                <Typography className={classes.text}>{content}</Typography>
              </Grid>
            </Grid>
          ))}
        </Grid>
      </Grid>
    </>
  );
}
