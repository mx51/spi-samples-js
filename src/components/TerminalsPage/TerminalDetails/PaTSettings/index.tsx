import React from 'react';
import Box from '@material-ui/core/Box';
import Grid from '@material-ui/core/Grid';
import Switch from '@material-ui/core/Switch';
import Typography from '@material-ui/core/Typography';
import { payAtTableFields, payAtTableOptions } from '../../../../definitions/constants/terminalConfigs';
import CustomTextField from '../../../CustomTextField';
import useStyles from './index.styles';

export default function PaTSettings(): React.ReactElement {
  const classes = useStyles();

  return (
    <>
      <Grid container className={classes.sectionSpacing}>
        <Grid item md={4} xs={12}>
          <Typography>Settings</Typography>
        </Grid>
        <Grid item md={8} xs={12}>
          {payAtTableOptions.map(({ name, title, subtitle }) => (
            <Box alignItems="center" className={classes.switchBox} display="flex" key={name}>
              <Typography align="right">
                <Switch name={name} inputProps={{ 'aria-label': title }} />
              </Typography>
              <Box display="flex" flexDirection="column" marginLeft={2}>
                <Typography variant="h5">{title}</Typography>
                <Typography variant="inherit">{subtitle}</Typography>
              </Box>
            </Box>
          ))}
        </Grid>
      </Grid>
      <Grid container className={classes.sectionSpacing}>
        <Grid item md={4} xs={12}>
          <Typography>Customisation</Typography>
        </Grid>
        <Grid item md={8} xs={12}>
          {payAtTableFields.map(({ helperText, id, label }) => (
            <CustomTextField
              key={id}
              fullWidth
              helperText={helperText}
              dataTestId={id}
              label={label}
              margin="dense"
              value=""
              variant="outlined"
            />
          ))}
        </Grid>
      </Grid>
    </>
  );
}
