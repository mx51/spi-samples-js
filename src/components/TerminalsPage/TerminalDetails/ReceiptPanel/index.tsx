import React from 'react';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import useStyles from './index.styles';

export default function ReceiptPanel(): React.ReactElement {
  const classes = useStyles();
  return (
    <Grid className={classes.root}>
      <Typography variant="h6" component="h1">
        Receipt
      </Typography>
    </Grid>
  );
}
