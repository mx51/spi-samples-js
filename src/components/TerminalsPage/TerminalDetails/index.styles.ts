import { createStyles, makeStyles, Theme } from '@material-ui/core';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      '& .MuiTabs-indicator': {
        backgroundColor: theme.palette.primary.main,
      },
    },
    tabs: {
      borderBottom: `1px solid ${theme.palette.secondary.main}`,
    },
  })
);

export default useStyles;
