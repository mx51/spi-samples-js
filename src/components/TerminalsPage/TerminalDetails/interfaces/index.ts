import React from 'react';
import { ITerminalProps } from '../../../../redux/reducers/TerminalSlice/interfaces';

export interface IReceiptToggles {
  settlement: boolean;
  settlementEnquiry: boolean;
}

export interface TabPanelProps {
  children: React.ReactNode;
  index: number;
  subtitle: string;
  title: string;
  value: number;
  receiptToggle?: IReceiptToggles;
  terminal?: ITerminalProps | null;
}

export interface ITerminalConfigurationConstants {
  title: string;
  content: string;
}

export interface IAboutTerminal {
  receiptToggle: IReceiptToggles;
  setReceiptToggle: (receiptToggle: IReceiptToggles) => void;
  terminal: ITerminalProps;
}
