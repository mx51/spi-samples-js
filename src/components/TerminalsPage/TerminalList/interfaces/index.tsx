export interface ITerminalRecord {
  posId: string;
  pairingStatus: PairingStatus;
  deviceAddress: string;
  serialNumber: string;
  status: string;
}
export interface ITerminalsRecords {
  terminals: Array<ITerminalRecord>;
}

export enum PairingStatus {
  Connecting,
  Connected,
  Disconnected,
  Idle,
}
