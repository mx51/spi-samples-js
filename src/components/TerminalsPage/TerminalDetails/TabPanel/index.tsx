import React from 'react';
import { Container } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { TxFlowState } from '../../../../definitions/constants/terminalConfigs';
import { useAppSelector } from '../../../../redux/hooks';
import { selectedShowFlowPanel } from '../../../../redux/reducers/CommonSlice/commonSliceSelectors';
import {
  terminalTxFlowFinishedTracker,
  terminalTxFlowReceipt,
  terminalTxFlowReceiptContent,
  terminalTxFlowSuccessTracker,
} from '../../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import FlowPanel from '../../../FlowPanel';
import PairFlow from '../../../FlowPanel/PairFlow';
import ReceiptPanel from '../../../ReceiptPanel';
import { TabPanelProps } from '../interfaces';
import useStyles from './index.styles';

export default function TabPanel({
  children,
  index,
  subtitle,
  title,
  value,
  receiptToggle,
  terminal,
}: TabPanelProps): React.ReactElement {
  const showFlowPanel = useAppSelector(selectedShowFlowPanel);
  const classes = useStyles({ showFlowPanel });

  const txFlowReceiptContent = useAppSelector(terminalTxFlowReceiptContent(terminal?.id as string));
  const txFlowReceipt = useAppSelector(terminalTxFlowReceipt(terminal?.id as string));
  const txFlowFinishedState = useAppSelector(terminalTxFlowFinishedTracker(terminal?.id as string));
  const txFlowSuccessState = useAppSelector(terminalTxFlowSuccessTracker(terminal?.id as string));

  const ReceiptContentDOM = () => (
    <>
      {!txFlowFinishedState && txFlowSuccessState === TxFlowState.Unknown && (
        <pre className={classes.preContent}>Settle Initiated. Will be updated with Progress.</pre>
      )}
      {(txFlowSuccessState === TxFlowState.Failed || txFlowReceipt?.errorReason) && txFlowFinishedState && (
        <pre className={classes.preContent}>
          Could not initiate settle: <strong>{txFlowReceipt?.errorReason}</strong>. Please Retry.
        </pre>
      )}
      {txFlowFinishedState && txFlowSuccessState === TxFlowState.Success && txFlowReceiptContent && (
        <pre className={classes.preContent}>{txFlowReceiptContent}</pre>
      )}
    </>
  );

  return (
    <>
      {value === index && (
        <div className={classes.root}>
          <div className={showFlowPanel ? `${classes.panel} ${classes.panelShift}` : classes.panel}>
            <Container maxWidth="lg">
              <Grid className={classes.detailsPanelContainer}>
                <div
                  className={
                    receiptToggle?.settlement || receiptToggle?.settlementEnquiry
                      ? `${classes.detailsPanel} ${classes.detailsPanelShift}`
                      : classes.detailsPanel
                  }
                >
                  <Grid container>
                    <Grid item className={classes.tabPanelContainer}>
                      <Grid container direction="column" justifyContent="space-between" spacing={1}>
                        <Typography variant="h6" component="h1">
                          {title}
                        </Typography>
                        <Typography className={classes.text}>{subtitle}</Typography>
                      </Grid>
                    </Grid>
                  </Grid>
                  <Grid>{children}</Grid>
                </div>
                <Grid
                  className={
                    receiptToggle?.settlement || receiptToggle?.settlementEnquiry
                      ? classes.receiptPanelOpened
                      : classes.receiptPanel
                  }
                >
                  {receiptToggle?.settlement && (
                    <ReceiptPanel title="Settlement Receipt" css={classes.receiptBoxWrapper}>
                      <ReceiptContentDOM />
                    </ReceiptPanel>
                  )}
                  {receiptToggle?.settlementEnquiry && (
                    <ReceiptPanel title="Settlement Enquiry Receipt" css={classes.receiptBoxWrapper}>
                      <ReceiptContentDOM />
                    </ReceiptPanel>
                  )}
                </Grid>
              </Grid>
            </Container>
          </div>
          <FlowPanel>
            <PairFlow terminal={terminal} />
          </FlowPanel>
        </div>
      )}
    </>
  );
}
