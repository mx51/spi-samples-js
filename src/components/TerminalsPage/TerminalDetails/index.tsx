import React, { useState } from 'react';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import { useLocation } from 'react-router-dom';
import { SPI_PAIR_STATUS } from '../../../definitions/constants/commonConfigs';
import { useAppSelector } from '../../../redux/hooks';
import { ITerminalProps } from '../../../redux/reducers/TerminalSlice/interfaces';
import { terminalInstance } from '../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import Layout from '../../Layout';
import NotFound from '../../NotFoundPage';
import AboutTerminal from './AboutTerminal';
import useStyles from './index.styles';
import PaTSettings from './PaTSettings';
import TabPanel from './TabPanel';

export default function TerminalDetails(): React.ReactElement {
  const [tabIndex, setTabIndex] = useState(0);
  const [receiptToggle, setReceiptToggle] = useState(false);

  const { pathname } = useLocation();
  const currentInstanceId = pathname?.split('/terminals/')[1];

  const currentTerminal = useAppSelector(terminalInstance(currentInstanceId)) as ITerminalProps;

  const classes = useStyles();

  const handleTabChange = (event: React.ReactNode, newValue: number) => {
    setTabIndex(newValue);
  };

  return (
    <Layout>
      {currentTerminal?.status === SPI_PAIR_STATUS.PairedConnected ? (
        <div className={classes.root}>
          <Tabs
            aria-label="terminal details tabs"
            className={classes.tabs}
            id="terminalDetailsTabs"
            onChange={handleTabChange}
            value={tabIndex}
          >
            <Tab id="aboutTerminalTab" label="About this terminal" />
            <Tab id="payAtTableTab" label="Pay at Table" />
          </Tabs>

          <TabPanel
            index={0}
            subtitle="View information about this terminal and the pairing configuration"
            title="About this terminal"
            value={tabIndex}
            receiptToggle={receiptToggle}
            terminal={currentTerminal}
          >
            <AboutTerminal
              receiptToggle={receiptToggle}
              setReceiptToggle={setReceiptToggle}
              terminal={currentTerminal as ITerminalProps}
            />
          </TabPanel>

          <TabPanel
            index={1}
            subtitle="Configure your Pay at Table settings for this terminal"
            title="Pay at Table"
            value={tabIndex}
            terminal={currentTerminal}
          >
            <PaTSettings />
          </TabPanel>
        </div>
      ) : (
        <NotFound />
      )}
    </Layout>
  );
}
