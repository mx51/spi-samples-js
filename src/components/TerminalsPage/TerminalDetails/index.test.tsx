import React from 'react';
import { cleanup, fireEvent } from '@testing-library/react';
import TerminalDetails from '.';
import { ITerminalState } from '../../../redux/reducers/TerminalSlice/interfaces';
import { terminalInstance } from '../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import spiService from '../../../services/spiService';
import mockWithRedux, { mockTerminalInstanceId, pairedMockTerminals } from '../../../utils/tests/common';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: `/terminals/${mockTerminalInstanceId}`,
  }),
}));

function setupContainer(terminals: ITerminalState = pairedMockTerminals) {
  spiService.spiHardwarePrinterAvailable = jest.fn().mockReturnValue(false);
  const customizedStore = {
    getState: () => ({
      common: {},
      pairForm: {
        serialNumber: {
          value: mockTerminalInstanceId,
          isValid: true,
        },
      },
      products: [],
      terminals,
    }),
    subscribe: jest.fn(),
    dispatch: jest.fn(),
  };

  return mockWithRedux(<TerminalDetails />, customizedStore);
}

describe('Test <TerminalDetails />', () => {
  let mockContainer: Any;

  beforeEach(() => {
    mockContainer = setupContainer();
  });

  afterEach(() => {
    jest.clearAllMocks();
    cleanup();
  });

  test('should match TerminalDetails snapshot tes', () => {
    // Arrange
    const subtitle = 'View information about this terminal and the pairing configuration';
    const useAppSelector = jest.fn();

    // Act
    useAppSelector.mockImplementation(terminalInstance(mockTerminalInstanceId));

    // Assert
    expect(mockContainer).toMatchSnapshot();
    expect(mockContainer.innerHTML.includes(subtitle)).toBeTruthy();
  });

  test('should should Page not found', () => {
    // Arrange
    const mockTerminalId = '234-234-234';
    const mockTitle = 'Page not found';
    const useAppSelector = jest.fn();

    const customizedStore = {
      getState: () => ({
        common: {},
        pairForm: {},
        products: [],
        terminals: {},
      }),
      subscribe: jest.fn(),
      dispatch: jest.fn(),
    };

    // Act
    const containerWrapper = mockWithRedux(<TerminalDetails />, customizedStore);
    useAppSelector.mockImplementation(terminalInstance(mockTerminalId));

    // Assert
    expect(containerWrapper.innerHTML.includes(mockTitle)).toBeTruthy();
  });

  test('should be toggle-able when clicks different tab', () => {
    // Arrange
    const payAtTableClickedDOM =
      '<button class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected" tabindex="0" type="button" role="tab" aria-selected="true" id="payAtTableTab"><span class="MuiTab-wrapper">Pay at Table</span><span class="MuiTouchRipple-root"></span></button>';
    const tabActiveProof = 'aria-selected="true"';

    const terminalDetailsTabsDOM = mockContainer.querySelector('#terminalDetailsTabs');
    const payAtTableTabDOM = mockContainer.querySelector('#payAtTableTab');

    // Act
    fireEvent.change(terminalDetailsTabsDOM);
    fireEvent.click(payAtTableTabDOM);

    // Assert
    expect(payAtTableClickedDOM.includes(tabActiveProof)).toBe(true);
  });
});
