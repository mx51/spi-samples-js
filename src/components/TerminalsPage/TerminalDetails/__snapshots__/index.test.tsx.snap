// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TerminalDetails /> should match TerminalDetails snapshot tes 1`] = `
<div>
  <div>
    <div
      class="makeStyles-root-3"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-navbar-4 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <div
            class="MuiBox-root MuiBox-root-9 makeStyles-navbar-6"
          >
            <div
              class="MuiBox-root MuiBox-root-10"
            >
              <button
                aria-label="menu"
                class="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-7 MuiIconButton-colorInherit MuiIconButton-edgeStart"
                data-test-id="navbarMenuIcon"
                name="menuIcon"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <svg>
                PrimaryLogoIcon.svg
              </svg>
            </div>
          </div>
        </div>
      </header>
    </div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="MuiTabs-root makeStyles-tabs-2"
        id="terminalDetailsTabs"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed"
          style="overflow: hidden;"
        >
          <div
            aria-label="terminal details tabs"
            class="MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected"
              id="aboutTerminalTab"
              role="tab"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                About this terminal
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit"
              id="payAtTableTab"
              role="tab"
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiTab-wrapper"
              >
                Pay at Table
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <span
            class="PrivateTabIndicator-root-11 PrivateTabIndicator-colorSecondary-13 MuiTabs-indicator"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="makeStyles-root-15 makeStyles-root-30"
      >
        <div
          class="makeStyles-panel-16"
        >
          <div
            class="MuiContainer-root MuiContainer-maxWidthLg"
          >
            <div
              class="MuiGrid-root makeStyles-detailsPanelContainer-23"
            >
              <div
                class="makeStyles-detailsPanel-24"
              >
                <div
                  class="MuiGrid-root MuiGrid-container"
                >
                  <div
                    class="MuiGrid-root makeStyles-tabPanelContainer-18 MuiGrid-item"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-justify-content-xs-space-between"
                    >
                      <h1
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        About this terminal
                      </h1>
                      <p
                        class="MuiTypography-root makeStyles-text-20 MuiTypography-body1"
                      >
                        View information about this terminal and the pairing configuration
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container"
                  >
                    <div
                      class="MuiGrid-root makeStyles-sectionSpacing-34 MuiGrid-item"
                    >
                      <h1
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        Terminal information
                      </h1>
                      <p
                        class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                      >
                        Information about this terminal
                      </p>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Pairing status
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <div
                            class="MuiBox-root MuiBox-root-43 makeStyles-statusBox-37"
                          >
                            <p
                              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
                            >
                              <svg
                                class="makeStyles-successIcon-42"
                              >
                                SuccessIcon.svg
                              </svg>
                            </p>
                            <div
                              class="MuiBox-root MuiBox-root-44"
                            >
                              <h5
                                class="MuiTypography-root makeStyles-statusTitle-38 MuiTypography-h5"
                              >
                                Connected
                              </h5>
                              <span
                                class="MuiTypography-root makeStyles-statusText-39"
                              >
                                Ready
                              </span>
                            </div>
                          </div>
                          <button
                            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-actionButton-36"
                            data-test-id="terminalDetailsSettlementBtn"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              Settlement
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                          <button
                            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-actionButton-36"
                            data-test-id="terminalDetailsSettlementEnquiryBtn"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              Settlement enquiry
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                          <a
                            aria-disabled="false"
                            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-actionButton-36"
                            data-test-id="terminalDetailsUnpairBtn"
                            href="/terminals"
                            role="button"
                            tabindex="0"
                          >
                            <span
                              class="MuiButton-label"
                            >
                              Unpair terminal
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </a>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Merchant ID
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            123456789
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Terminal ID
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            987654321
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Serial number
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            123-123-123
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Battery
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            40%
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Print customer EFTPOS receipt
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <span
                            class="MuiSwitch-root"
                          >
                            <span
                              aria-disabled="true"
                              class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-45 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-disabled-47 Mui-disabled Mui-disabled Mui-disabled"
                              tabindex="-1"
                            >
                              <span
                                class="MuiIconButton-label"
                              >
                                <input
                                  class="PrivateSwitchBase-input-48 MuiSwitch-input"
                                  disabled=""
                                  type="checkbox"
                                  value=""
                                />
                                <span
                                  class="MuiSwitch-thumb"
                                />
                              </span>
                            </span>
                            <span
                              class="MuiSwitch-track"
                            />
                          </span>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root makeStyles-sectionSpacing-34 MuiGrid-item"
                    >
                      <h1
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        Pairing configuration
                      </h1>
                      <p
                        class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                      >
                        To update the Pairing configuration you need to unpair this terminal using the Unpair button above
                      </p>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Payment provider
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            test
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            Configuration option
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            No
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            EFTPOS address
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            192.168.1.10
                          </p>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root makeStyles-detailRow-33 MuiGrid-container"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
                        >
                          <p
                            class="MuiTypography-root MuiTypography-body1"
                          >
                            POS ID
                          </p>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8"
                        >
                          <p
                            class="MuiTypography-root makeStyles-text-32 MuiTypography-body1"
                          >
                            test
                          </p>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root makeStyles-receiptPanel-26"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-49"
          data-test-id="flowPanel"
        >
          <div
            class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-50 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
            style="visibility: hidden; webkit-transform: translateX(1024px) translateX(0px); transform: translateX(1024px) translateX(0px);"
          >
            <div
              class="MuiBox-root MuiBox-root-54 makeStyles-flowBoxWrapper-51"
            >
              <div
                class="MuiBox-root MuiBox-root-55 makeStyles-flowBox-52"
              >
                <h1
                  class="MuiTypography-root MuiTypography-h6"
                >
                  Flow
                </h1>
                <pre
                  class="makeStyles-flowContent-53"
                >
                  
# ----------- STATUS -----------

# POSID: test
# EFTPOS: 192.168.1.10
# SPI STATUS: PairedConnected
# FLOW: null

# ------------------------------
# POS: v1.2.3 Spi: 1.2.2
  
                </pre>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
